### PR TITLE
add optional diagnostic thermodynamic covariance in the environment

### PR DESCRIPTION
**Pull request by yairchn**
This PR build's on @trontrytel's branch and making the diagnostic covariances in the environment optional. 
We should add vanilla+ test for this option

### CODE REVIEW

File: src/EDMF_functions.jl [hunk 1]
```diff
@@ -38,9 +38,16 @@ function compute_turbconv_tendencies!(
...
-    compute_en_tendencies!(edmf, grid, state, param_set, Val(:HQTcov), Val(:ρaHQTcov))
+
+    if edmf.thermo_covariance_model isa PrognosticThermoCovariances
+        compute_en_tendencies!(edmf, grid, state, param_set, Val(:Hvar), Val(:ρatke)) # TODO - hack!
+        compute_en_tendencies!(edmf, grid, state, param_set, Val(:QTvar), Val(:ρatke))
+        compute_en_tendencies!(edmf, grid, state, param_set, Val(:HQTcov), Val(:ρatke))
+    else
+        update_diagnostic_covariances!(edmf, grid, state, param_set, Val(:Hvar), Val(:ρatke)) # TODO - hack!
+        update_diagnostic_covariances!(edmf, grid, state, param_set, Val(:QTvar), Val(:ρatke))
+        update_diagnostic_covariances!(edmf, grid, state, param_set, Val(:HQTcov), Val(:ρatke))
```
Comment (charleskawczynski) [inline src/EDMF_functions.jl:49]: Why are we computing diagnostics in `compute_turbconv_tendencies!`?

File: src/EDMF_functions.jl [hunk 2]
```diff
@@ -246,10 +254,17 @@ function affect_filter!(
...
+        if edmf.thermo_covariance_model isa PrognosticThermoCovariances
+            prog_en.ρaHvar[k] = max(prog_en.ρaHvar[k], 0.0)
+            prog_en.ρaQTvar[k] = max(prog_en.ρaQTvar[k], 0.0)
+            prog_en.ρaHQTcov[k] = max(prog_en.ρaHQTcov[k], -sqrt(prog_en.ρaHvar[k] * prog_en.ρaQTvar[k]))
+            prog_en.ρaHQTcov[k] = min(prog_en.ρaHQTcov[k], sqrt(prog_en.ρaHvar[k] * prog_en.ρaQTvar[k]))
+        elseif edmf.thermo_covariance_model isa DiagnosticThermoCovariances
+            aux_en.Hvar[k] = max(aux_en.Hvar[k], 0.0)
+            aux_en.QTvar[k] = max(aux_en.QTvar[k], 0.0)
+            aux_en.HQTcov[k] = max(aux_en.HQTcov[k], -sqrt(aux_en.Hvar[k] * aux_en.QTvar[k]))
+            aux_en.HQTcov[k] = min(aux_en.HQTcov[k], sqrt(aux_en.Hvar[k] * aux_en.QTvar[k]))
```
Comment (charleskawczynski) [inline src/EDMF_functions.jl:266]: The filter is only for prognostic variables, not auxiliary.

File: src/EDMF_functions.jl [hunk 3]
```diff
@@ -293,9 +309,15 @@ function set_edmf_surface_bc(edmf::EDMFModel, grid::Grid, state::State, surf::Su
...
-    prog_en.ρaQTvar[kc_surf] = ρ0_ae * get_surface_variance(flux2 * α0LL, flux2 * α0LL, ustar, zLL, oblength)
-    prog_en.ρaHQTcov[kc_surf] = ρ0_ae * get_surface_variance(flux1 * α0LL, flux2 * α0LL, ustar, zLL, oblength)
+    if edmf.thermo_covariance_model isa PrognosticThermoCovariances
+        prog_en.ρaHvar[kc_surf] = ρ0_ae * get_surface_variance(flux1 * α0LL, flux1 * α0LL, ustar, zLL, oblength)
+        prog_en.ρaQTvar[kc_surf] = ρ0_ae * get_surface_variance(flux2 * α0LL, flux2 * α0LL, ustar, zLL, oblength)
+        prog_en.ρaHQTcov[kc_surf] = ρ0_ae * get_surface_variance(flux1 * α0LL, flux2 * α0LL, ustar, zLL, oblength)
+    elseif edmf.thermo_covariance_model isa DiagnosticThermoCovariances
+        aux_en.Hvar[kc_surf] = get_surface_variance(flux1 * α0LL, flux1 * α0LL, ustar, zLL, oblength)
+        aux_en.QTvar[kc_surf] = get_surface_variance(flux2 * α0LL, flux2 * α0LL, ustar, zLL, oblength)
+        aux_en.HQTcov[kc_surf] = get_surface_variance(flux1 * α0LL, flux2 * α0LL, ustar, zLL, oblength)
```
Comment (charleskawczynski) [inline src/EDMF_functions.jl:319]: Shouldn't this be in `update_aux!`?

**Comment by yairchn** [comment]
@charleskawczynski could you take look, its not working and I can't find why

**Review by trontrytel — APPROVED** [review]

**Review by charleskawczynski — APPROVED** [review]
Code changes look fine to me

**Pull request merged**